mesh/pmesh.cpp

```diff
@@ -5568,6 +5568,97 @@ void ParMesh::PrintSharedEntities(const char *fname_prefix) const
    }
 }
 
+void ParMesh::GetGlobalVertexIndices(Array<HYPRE_Int> &gi)
+{
+   H1_FECollection fec(1, Dim); // Order 1, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNV(),
+               "There should be a bijection between DOFs and vertices");
+
+   gi.SetSize(GetNV());
+
+   Array<int> dofs;
+   bool oneDof = true;
+   for (int i=0; i<GetNV(); ++i)
+   {
+      fespace.GetVertexDofs(i, dofs);
+      oneDof = oneDof && (dofs.Size() == 1);
+      gi[i] = fespace.GetGlobalTDofNumber(dofs[0]);
+   }
+
+   MFEM_VERIFY(oneDof, "Each vertex should have exactly one DOF");
+}
+
+void ParMesh::GetGlobalEdgeIndices(Array<HYPRE_Int> &gi)
+{
+   ND_FECollection fec(1, Dim); // Order 1, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNEdges(),
+               "There should be a bijection between DOFs and edges");
+
+   gi.SetSize(GetNEdges());
+
+   Array<int> dofs;
+   bool oneDof = true;
+   for (int i=0; i<GetNEdges(); ++i)
+   {
+      fespace.GetEdgeDofs(i, dofs);
+      oneDof = oneDof && (dofs.Size() == 1);
+      const int ldof = (dofs[0] >= 0) ? dofs[0] : -1 - dofs[0];
+      gi[i] = fespace.GetGlobalTDofNumber(ldof);
+   }
+
+   MFEM_VERIFY(oneDof, "Each edge should have exactly one DOF");
+}
+
+void ParMesh::GetGlobalFaceIndices(Array<HYPRE_Int> &gi)
+{
+   MFEM_VERIFY(Dim == 3, "GetGlobalFaceIndices can be called only in 3D");
+   RT_FECollection fec(0, Dim); // Order 0, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNFaces(),
+               "There should be a bijection between DOFs and faces");
+
+   gi.SetSize(GetNFaces());
+
+   Array<int> dofs;
+   bool oneDof = true;
+   for (int i=0; i<GetNFaces(); ++i)
+   {
+      fespace.GetFaceDofs(i, dofs);
+      oneDof = oneDof && (dofs.Size() == 1);
+      const int ldof = (dofs[0] >= 0) ? dofs[0] : -1 - dofs[0];
+      gi[i] = fespace.GetGlobalTDofNumber(ldof);
+   }
+
+   MFEM_VERIFY(oneDof, "Each face should have exactly one DOF");
+}
+
+void ParMesh::GetGlobalElementIndices(Array<HYPRE_Int> &gi)
+{
```

Comment:
`ParMesh` already has a `GetGlobalElementNum` method. So maybe we don't need this, or it can actually get used here to construct the array of element indices without the need for an L2 space. 
In addition there is a `GetLocalElementNum(long global_element_num)`.  You think adding the global to local map for vertices, edges and faces would be useful too? I am thinking...why not?

Reply:
Yes, I agree that the case of elements is simpler than faces, edges, or vertices, in that each element is on exactly one process, so global indices can be determined by the number of elements on each process. So we could use `ComputeGlobalElementOffset` to set the array of indices, rather than an L2 space. I will make this change. The inverse map from global to local indices could be constructed with an `std::map<HYPRE_Int, int>` for vertices, edges, and faces. This seems simple enough for the user to create if needed, and I'm not sure it belongs in `ParMesh` as a function to return the `std::map` for an input `Array` of global indices (unless you see a better way).